linker/descriptors.go

```diff
@@ -793,42 +834,44 @@ func (e *enumDescriptor) Options() protoreflect.ProtoMessage {
 }
 
 func (e *enumDescriptor) Values() protoreflect.EnumValueDescriptors {
-	// Unlike all other elements, the fully-qualified name of enum values
-	// is NOT scoped to their parent element (the enum), but rather to
-	// the enum's parent element. This follows C++ scoping rules for
-	// enum values.
-	prefix := strings.TrimSuffix(e.fqn, e.proto.GetName())
-	return &enValDescriptors{file: e.file, parent: e, vals: e.proto.GetValue(), prefix: prefix}
+	return &e.values
 }
 
 func (e *enumDescriptor) ReservedNames() protoreflect.Names {
-	return names{s: e.proto.ReservedName}
+	return e.rsvdNames
 }
 
 func (e *enumDescriptor) ReservedRanges() protoreflect.EnumRanges {
-	return enumRanges{s: e.proto.ReservedRange}
+	return e.rsvdRanges
 }
 
 type enumRanges struct {
 	protoreflect.EnumRanges
-	s []*descriptorpb.EnumDescriptorProto_EnumReservedRange
+	ranges [][2]protoreflect.EnumNumber
+}
+
+func createEnumRanges(rangeProtos []*descriptorpb.EnumDescriptorProto_EnumReservedRange) enumRanges {
+	ranges := make([][2]protoreflect.EnumNumber, len(rangeProtos))
+	for i, r := range rangeProtos {
+		ranges[i] = [2]protoreflect.EnumNumber{
+			protoreflect.EnumNumber(r.GetStart()),
+			protoreflect.EnumNumber(r.GetEnd()),
+		}
+	}
+	return enumRanges{ranges: ranges}
 }
 
 func (e enumRanges) Len() int {
-	return len(e.s)
+	return len(e.ranges)
 }
 
 func (e enumRanges) Get(i int) [2]protoreflect.EnumNumber {
-	r := e.s[i]
-	return [2]protoreflect.EnumNumber{
-		protoreflect.EnumNumber(r.GetStart()),
-		protoreflect.EnumNumber(r.GetEnd()),
-	}
+	return e.ranges[i]
 }
 
 func (e enumRanges) Has(n protoreflect.EnumNumber) bool {
-	for _, r := range e.s {
-		if r.GetStart() <= int32(n) && r.GetEnd() >= int32(n) {
+	for _, r := range e.ranges {
+		if r[0] <= n && r[0] >= n {
```

Comment:
```suggestion
		if r[0] <= n && r[1] >= n {
```

I think this logic isn't correct - `r[1]` should be the equivalent of `r.GetEnd()` from before, right?

Reply:
You are right again. Thanks for spotting that!